build: Post-pr23489 small cleanups

hebasto opened the pull request:
This PR suggests small cleanups as follow ups of bitcoin/bitcoin#23489.

hebasto committed:
build, qt: Drop unused CROSS_COMPILE variable for Android builds

Since Qt 5.14.0 the CROSS_COMPILE variable always being overridden by
the Qt build system internally.

See upstream commit: c28b881c98fadcd3415370fad2525b558f6b03e4

hebasto committed:
build, refactor: Re-use `qt_lib_suffix` variable

hebasto committed:
build: Use config_opts instead of passing a compiler flag directly

This approach is safer as it allows the configure script to check the
availability of a compiler flag.

hebasto commented:
cc @icota @fanquake @jarolrod

hebasto commented:
#### Guix builds:
```
$ find guix-build-$(git rev-parse --short=12 HEAD)/output/ -type f -print0 | env LC_ALL=C sort -z | xargs -r0 sha256sum
83bd24144872ef0c029a4fcca6e570a690cbf0c9964dd13270d185d4bdded765  guix-build-6bf6904e3f17/output/aarch64-linux-gnu/SHA256SUMS.part
1eb263b4930cf93a3c7f44753afc6ceb6e445763bf8faad57656ea96bf454346  guix-build-6bf6904e3f17/output/aarch64-linux-gnu/bitcoin-6bf6904e3f17-aarch64-linux-gnu-debug.tar.gz
1eb81d905f125b48d70072550ffdce5c4787c0a3dbf8bcbbf900b2426d63baa6  guix-build-6bf6904e3f17/output/aarch64-linux-gnu/bitcoin-6bf6904e3f17-aarch64-linux-gnu.tar.gz
4ac732f23162c2f79933909c2106967b3c4747af58efc6d1dbf5a111e562312f  guix-build-6bf6904e3f17/output/arm-linux-gnueabihf/SHA256SUMS.part
c1f35ced0fccc5744575f0e0dcfd57abf4a94e709e9c53e95f1a31353f2c9e8c  guix-build-6bf6904e3f17/output/arm-linux-gnueabihf/bitcoin-6bf6904e3f17-arm-linux-gnueabihf-debug.tar.gz
a7cb9e960ffcf1405e8a318967eb2706d61bbcb66f2c5cf0fec9de493df0785e  guix-build-6bf6904e3f17/output/arm-linux-gnueabihf/bitcoin-6bf6904e3f17-arm-linux-gnueabihf.tar.gz
0c08877c5b4e5b0896d4cd9eecee5cf9d66fcec253547c3ab1154774f14d8c72  guix-build-6bf6904e3f17/output/dist-archive/bitcoin-6bf6904e3f17.tar.gz
229c4e817b5ca97ff1a3dbf913b22256337d98594128062b753561070f4e9354  guix-build-6bf6904e3f17/output/powerpc64-linux-gnu/SHA256SUMS.part
cc4bddd5ff9e501d6324a48dce01bfb43dd7ab190125f7bfe00899c3cbe786f0  guix-build-6bf6904e3f17/output/powerpc64-linux-gnu/bitcoin-6bf6904e3f17-powerpc64-linux-gnu-debug.tar.gz
caab6d722aaf57335109260c26cb5eb1d2b8f961c6e3580e91acf9766a065e93  guix-build-6bf6904e3f17/output/powerpc64-linux-gnu/bitcoin-6bf6904e3f17-powerpc64-linux-gnu.tar.gz
2d7c4a86e0c5576d78cc74ccd0ae7c71676f4aa4879f25cdd970de217bcddbdc  guix-build-6bf6904e3f17/output/powerpc64le-linux-gnu/SHA256SUMS.part
cb85ad79e260a722a575d58ae6841ba5ddab3e13854c8574c69e76e1229e4527  guix-build-6bf6904e3f17/output/powerpc64le-linux-gnu/bitcoin-6bf6904e3f17-powerpc64le-linux-gnu-debug.tar.gz
94d6defb1c4786a8cb0ba414efa84476228e432a653d292c6b437e087ae651bf  guix-build-6bf6904e3f17/output/powerpc64le-linux-gnu/bitcoin-6bf6904e3f17-powerpc64le-linux-gnu.tar.gz
96dfc260ba568b60c2a733ed6cac49eebbe1be1a81470c6f278588d310f3f4de  guix-build-6bf6904e3f17/output/riscv64-linux-gnu/SHA256SUMS.part
566720e6f8fa9dad8802bf640424bbf5a5673a4e3080e5d27f078ac18ed5bb59  guix-build-6bf6904e3f17/output/riscv64-linux-gnu/bitcoin-6bf6904e3f17-riscv64-linux-gnu-debug.tar.gz
dcc11ad377567a98d7a1680da14971a9eba29ebdfde52cfeb1684aba08b37793  guix-build-6bf6904e3f17/output/riscv64-linux-gnu/bitcoin-6bf6904e3f17-riscv64-linux-gnu.tar.gz
f0ca32522142986246e036f7553619d6e2ce8da54c7f89f441d10ece83e6f68e  guix-build-6bf6904e3f17/output/x86_64-apple-darwin19/SHA256SUMS.part
c3d50cba0f03a94a55e16192a040092dde7b360d78ea9003ce97afa21e7dc83e  guix-build-6bf6904e3f17/output/x86_64-apple-darwin19/bitcoin-6bf6904e3f17-osx-unsigned.dmg
8e044c5d3c5b0985bb811ede01b3ec9c22c06ccd4d64889d52c4a7bfef09201d  guix-build-6bf6904e3f17/output/x86_64-apple-darwin19/bitcoin-6bf6904e3f17-osx-unsigned.tar.gz
1b3036ad7f8b9f2d201141e5993642e2e9363b801f46f3ca2cac4522310ec028  guix-build-6bf6904e3f17/output/x86_64-apple-darwin19/bitcoin-6bf6904e3f17-osx64.tar.gz
f23c169b2bb42ac4ba4c5df92b4a1cf3b9de8db38d6fed12e7187fafd38865a0  guix-build-6bf6904e3f17/output/x86_64-linux-gnu/SHA256SUMS.part
87e9faa60f1b84efa081ac6761c692190fb9103b0f638e9ad1fe3fb328cd7733  guix-build-6bf6904e3f17/output/x86_64-linux-gnu/bitcoin-6bf6904e3f17-x86_64-linux-gnu-debug.tar.gz
f5693ee99ac613892736a3b0204f7bd43600d5fd40a2a75fce4792d852ba11ca  guix-build-6bf6904e3f17/output/x86_64-linux-gnu/bitcoin-6bf6904e3f17-x86_64-linux-gnu.tar.gz
5923665ba701cb2dbcb940dda3bf461337ce444e196ae920ef3416f92b827dfc  guix-build-6bf6904e3f17/output/x86_64-w64-mingw32/SHA256SUMS.part
4bed0a7b41e693ba233790686040c0a585d278b5af6c412e4bb66315ae74c54f  guix-build-6bf6904e3f17/output/x86_64-w64-mingw32/bitcoin-6bf6904e3f17-win-unsigned.tar.gz
9209e4d08b31d9484714a662d9e2a6504ca9c41600403f565062aa5bf6b38ba5  guix-build-6bf6904e3f17/output/x86_64-w64-mingw32/bitcoin-6bf6904e3f17-win64-debug.zip
e56cce3a30b5e4697e09907443ea02099ca4a4587f2c67e47b0832140550cd92  guix-build-6bf6904e3f17/output/x86_64-w64-mingw32/bitcoin-6bf6904e3f17-win64-setup-unsigned.exe
221a8350d4c1dc3334c787182825274c23f0ec7d97025187d6f34e4c72870b92  guix-build-6bf6904e3f17/output/x86_64-w64-mingw32/bitcoin-6bf6904e3f17-win64.zip
```

icota commented:
I'm surprised `i686-linux-android` ABI is still supported. I would love to drop it but the variable renaming and PIC fix of this PR stand by themselves.

utACK https://github.com/bitcoin/bitcoin/pull/23675/commits/6bf6904e3f17c6aab19a25026d1e4e8aeaa71a35

fanquake commented:
Changes look ok, but I'm seeing a mismatch in the Guix build:
```bash
bash-5.1# find guix-build-$(git rev-parse --short=12 HEAD)/output/ -type f -print0 | env LC_ALL=C sort -z | xargs -r0 sha256sum
83bd24144872ef0c029a4fcca6e570a690cbf0c9964dd13270d185d4bdded765  guix-build-6bf6904e3f17/output/aarch64-linux-gnu/SHA256SUMS.part
1eb263b4930cf93a3c7f44753afc6ceb6e445763bf8faad57656ea96bf454346  guix-build-6bf6904e3f17/output/aarch64-linux-gnu/bitcoin-6bf6904e3f17-aarch64-linux-gnu-debug.tar.gz
1eb81d905f125b48d70072550ffdce5c4787c0a3dbf8bcbbf900b2426d63baa6  guix-build-6bf6904e3f17/output/aarch64-linux-gnu/bitcoin-6bf6904e3f17-aarch64-linux-gnu.tar.gz
4ac732f23162c2f79933909c2106967b3c4747af58efc6d1dbf5a111e562312f  guix-build-6bf6904e3f17/output/arm-linux-gnueabihf/SHA256SUMS.part
c1f35ced0fccc5744575f0e0dcfd57abf4a94e709e9c53e95f1a31353f2c9e8c  guix-build-6bf6904e3f17/output/arm-linux-gnueabihf/bitcoin-6bf6904e3f17-arm-linux-gnueabihf-debug.tar.gz
a7cb9e960ffcf1405e8a318967eb2706d61bbcb66f2c5cf0fec9de493df0785e  guix-build-6bf6904e3f17/output/arm-linux-gnueabihf/bitcoin-6bf6904e3f17-arm-linux-gnueabihf.tar.gz
0c08877c5b4e5b0896d4cd9eecee5cf9d66fcec253547c3ab1154774f14d8c72  guix-build-6bf6904e3f17/output/dist-archive/bitcoin-6bf6904e3f17.tar.gz
229c4e817b5ca97ff1a3dbf913b22256337d98594128062b753561070f4e9354  guix-build-6bf6904e3f17/output/powerpc64-linux-gnu/SHA256SUMS.part
cc4bddd5ff9e501d6324a48dce01bfb43dd7ab190125f7bfe00899c3cbe786f0  guix-build-6bf6904e3f17/output/powerpc64-linux-gnu/bitcoin-6bf6904e3f17-powerpc64-linux-gnu-debug.tar.gz
caab6d722aaf57335109260c26cb5eb1d2b8f961c6e3580e91acf9766a065e93  guix-build-6bf6904e3f17/output/powerpc64-linux-gnu/bitcoin-6bf6904e3f17-powerpc64-linux-gnu.tar.gz
2d7c4a86e0c5576d78cc74ccd0ae7c71676f4aa4879f25cdd970de217bcddbdc  guix-build-6bf6904e3f17/output/powerpc64le-linux-gnu/SHA256SUMS.part
cb85ad79e260a722a575d58ae6841ba5ddab3e13854c8574c69e76e1229e4527  guix-build-6bf6904e3f17/output/powerpc64le-linux-gnu/bitcoin-6bf6904e3f17-powerpc64le-linux-gnu-debug.tar.gz
94d6defb1c4786a8cb0ba414efa84476228e432a653d292c6b437e087ae651bf  guix-build-6bf6904e3f17/output/powerpc64le-linux-gnu/bitcoin-6bf6904e3f17-powerpc64le-linux-gnu.tar.gz
6bb1a7a7ba0eac335c43e3f84b1330f7683ba97c7b88ea026bf4143b56a7d32f  guix-build-6bf6904e3f17/output/riscv64-linux-gnu/SHA256SUMS.part
566720e6f8fa9dad8802bf640424bbf5a5673a4e3080e5d27f078ac18ed5bb59  guix-build-6bf6904e3f17/output/riscv64-linux-gnu/bitcoin-6bf6904e3f17-riscv64-linux-gnu-debug.tar.gz
d0df0e876ec869b2f6ff2f9758d600989ec82f4e2e02b06d62a6e779ecef0bfb  guix-build-6bf6904e3f17/output/riscv64-linux-gnu/bitcoin-6bf6904e3f17-riscv64-linux-gnu.tar.gz
f0ca32522142986246e036f7553619d6e2ce8da54c7f89f441d10ece83e6f68e  guix-build-6bf6904e3f17/output/x86_64-apple-darwin19/SHA256SUMS.part
c3d50cba0f03a94a55e16192a040092dde7b360d78ea9003ce97afa21e7dc83e  guix-build-6bf6904e3f17/output/x86_64-apple-darwin19/bitcoin-6bf6904e3f17-osx-unsigned.dmg
8e044c5d3c5b0985bb811ede01b3ec9c22c06ccd4d64889d52c4a7bfef09201d  guix-build-6bf6904e3f17/output/x86_64-apple-darwin19/bitcoin-6bf6904e3f17-osx-unsigned.tar.gz
1b3036ad7f8b9f2d201141e5993642e2e9363b801f46f3ca2cac4522310ec028  guix-build-6bf6904e3f17/output/x86_64-apple-darwin19/bitcoin-6bf6904e3f17-osx64.tar.gz
f23c169b2bb42ac4ba4c5df92b4a1cf3b9de8db38d6fed12e7187fafd38865a0  guix-build-6bf6904e3f17/output/x86_64-linux-gnu/SHA256SUMS.part
87e9faa60f1b84efa081ac6761c692190fb9103b0f638e9ad1fe3fb328cd7733  guix-build-6bf6904e3f17/output/x86_64-linux-gnu/bitcoin-6bf6904e3f17-x86_64-linux-gnu-debug.tar.gz
f5693ee99ac613892736a3b0204f7bd43600d5fd40a2a75fce4792d852ba11ca  guix-build-6bf6904e3f17/output/x86_64-linux-gnu/bitcoin-6bf6904e3f17-x86_64-linux-gnu.tar.gz
5923665ba701cb2dbcb940dda3bf461337ce444e196ae920ef3416f92b827dfc  guix-build-6bf6904e3f17/output/x86_64-w64-mingw32/SHA256SUMS.part
4bed0a7b41e693ba233790686040c0a585d278b5af6c412e4bb66315ae74c54f  guix-build-6bf6904e3f17/output/x86_64-w64-mingw32/bitcoin-6bf6904e3f17-win-unsigned.tar.gz
9209e4d08b31d9484714a662d9e2a6504ca9c41600403f565062aa5bf6b38ba5  guix-build-6bf6904e3f17/output/x86_64-w64-mingw32/bitcoin-6bf6904e3f17-win64-debug.zip
e56cce3a30b5e4697e09907443ea02099ca4a4587f2c67e47b0832140550cd92  guix-build-6bf6904e3f17/output/x86_64-w64-mingw32/bitcoin-6bf6904e3f17-win64-setup-unsigned.exe
221a8350d4c1dc3334c787182825274c23f0ec7d97025187d6f34e4c72870b92  guix-build-6bf6904e3f17/output/x86_64-w64-mingw32/bitcoin-6bf6904e3f17-win64.zip
```

hebasto commented:
Rebased 6bf6904e3f17c6aab19a25026d1e4e8aeaa71a35 -> 15676734ab90c9abe3c2e801197ba8045265263d ([pr23675.01](https://github.com/hebasto/bitcoin/commits/pr23675.01) -> [pr23675.02](https://github.com/hebasto/bitcoin/commits/pr23675.02)) due to the conflict with #23593.

hebasto commented:
#### Guix builds:
```
$ find guix-build-$(git rev-parse --short=12 HEAD)/output/ -type f -print0 | env LC_ALL=C sort -z | xargs -r0 sha256sum
4eccf22f62b4a4d50c988461ed5958f5629fcadd725e80350662c223eb3a1edb  guix-build-15676734ab90/output/aarch64-linux-gnu/SHA256SUMS.part
8defa41a3ea1adb8ccffb8a2b70fbcc724e13c8305e0bc9b4f3c5bfa1695d486  guix-build-15676734ab90/output/aarch64-linux-gnu/bitcoin-15676734ab90-aarch64-linux-gnu-debug.tar.gz
9b660f1c2e717c24e6449a9ae22e898423bb50b0e67f6d22951f4cb25c0171fc  guix-build-15676734ab90/output/aarch64-linux-gnu/bitcoin-15676734ab90-aarch64-linux-gnu.tar.gz
527e9279f4a57ea8069f5657f79bd48459b019df789c9bb91b910709e05db93d  guix-build-15676734ab90/output/arm-linux-gnueabihf/SHA256SUMS.part
f7b2ce2c4b39719bc76d1d4c74449eaecbeeeab750dbe03acecf986655b52bb0  guix-build-15676734ab90/output/arm-linux-gnueabihf/bitcoin-15676734ab90-arm-linux-gnueabihf-debug.tar.gz
e5462f0d47e4d1735b15ed191a56b929f179e8859e18453765eab576071b98a4  guix-build-15676734ab90/output/arm-linux-gnueabihf/bitcoin-15676734ab90-arm-linux-gnueabihf.tar.gz
ac3efea92545063dc5a2271c751b2fb600a82d8ecaf27b578f3266e17788925c  guix-build-15676734ab90/output/dist-archive/bitcoin-15676734ab90.tar.gz
59622c0793ed73c08e28cd85aa3f8bf65fa76f50cd59096eb29bc7142f4d5d6a  guix-build-15676734ab90/output/powerpc64-linux-gnu/SHA256SUMS.part
37eb1ddad45297a6d9b0fb8567ccd8d59cfb70ca039a4cbe2e4c93f246386482  guix-build-15676734ab90/output/powerpc64-linux-gnu/bitcoin-15676734ab90-powerpc64-linux-gnu-debug.tar.gz
4082532f47fcb3f4c9b76783eb5f34406a2c19bc77ab8b8b67fccd5f361730de  guix-build-15676734ab90/output/powerpc64-linux-gnu/bitcoin-15676734ab90-powerpc64-linux-gnu.tar.gz
d324530b017bc4fb62f33c3a56740fde081f394795f39014055da9b2daa40264  guix-build-15676734ab90/output/powerpc64le-linux-gnu/SHA256SUMS.part
a239ff8e07f599db195d9f9898bd71128294c9ed1ac3bff8d047050fcffb7075  guix-build-15676734ab90/output/powerpc64le-linux-gnu/bitcoin-15676734ab90-powerpc64le-linux-gnu-debug.tar.gz
f03f9987b8f5c87c51be428e6aa9a648694837d07f22f4ffd7dfc1fdc01ef8d3  guix-build-15676734ab90/output/powerpc64le-linux-gnu/bitcoin-15676734ab90-powerpc64le-linux-gnu.tar.gz
3279932a725b3b3353f8326cb4241b56a4f1d49e82c399b15cf17ecf40187e62  guix-build-15676734ab90/output/riscv64-linux-gnu/SHA256SUMS.part
77aefc989fcc2b7fc5bf213e79ac564f1b14b724a4f0166103d5d6007dae6421  guix-build-15676734ab90/output/riscv64-linux-gnu/bitcoin-15676734ab90-riscv64-linux-gnu-debug.tar.gz
c718e8df35adce73e2c7c6ea1d56197e8452c61f27a209d5f0f2095ae6c0a256  guix-build-15676734ab90/output/riscv64-linux-gnu/bitcoin-15676734ab90-riscv64-linux-gnu.tar.gz
4c1325be05ac5519e3d6a9230d74d84237e872cf4885c0f247da4ab24bdb0bf2  guix-build-15676734ab90/output/x86_64-apple-darwin/SHA256SUMS.part
cf2e123c6337055190a45792768a575bf093d188bb7f7c6dc7ad3e8ea85f1689  guix-build-15676734ab90/output/x86_64-apple-darwin/bitcoin-15676734ab90-osx-unsigned.dmg
cd2685a6a1936ea97d8e65755afb1203ce57073ce0cfbf2b38a9a4f4edd0a721  guix-build-15676734ab90/output/x86_64-apple-darwin/bitcoin-15676734ab90-osx-unsigned.tar.gz
0842f74a8f25dfba263b59b25cc0221d2c5e5a0ee6cb1d1e2655651acbf6dac7  guix-build-15676734ab90/output/x86_64-apple-darwin/bitcoin-15676734ab90-osx64.tar.gz
27945f4d243f7110b87c6f061bd3a54367b6faf5bcc36ab81012bd407f70e521  guix-build-15676734ab90/output/x86_64-linux-gnu/SHA256SUMS.part
e58dd9f2053e9a1da906ed23a52ecb1648b083bc454016ec75dc8d04846910a2  guix-build-15676734ab90/output/x86_64-linux-gnu/bitcoin-15676734ab90-x86_64-linux-gnu-debug.tar.gz
2034e7947db0e7f57e6e8b2eb47a53167421bd9ea939f47e1cfcf8ac33015b7c  guix-build-15676734ab90/output/x86_64-linux-gnu/bitcoin-15676734ab90-x86_64-linux-gnu.tar.gz
b9b33044808a639937119cc82248b4e61e5c08022f8fd34888c5128bb8fcce0b  guix-build-15676734ab90/output/x86_64-w64-mingw32/SHA256SUMS.part
f44a7d90af8c05a9ff66782c7a3f926627746775893f2200a0359021b9b2a950  guix-build-15676734ab90/output/x86_64-w64-mingw32/bitcoin-15676734ab90-win-unsigned.tar.gz
1be52ccd44b536bbf4d43cea9fa06b4ae0c6ee9dde798a23649d387dfc8c9750  guix-build-15676734ab90/output/x86_64-w64-mingw32/bitcoin-15676734ab90-win64-debug.zip
e0f1a64c4d8ab62e3eee5ff87815d5b846a8b242713e44579fe658089fdbc19c  guix-build-15676734ab90/output/x86_64-w64-mingw32/bitcoin-15676734ab90-win64-setup-unsigned.exe
ab8af1d9fd6bc379d85f9b925ff100e3b51a48ae8866402ccd5b7f00b5c8ae5c  guix-build-15676734ab90/output/x86_64-w64-mingw32/bitcoin-15676734ab90-win64.zip
```

laanwj commented:
What is `i686-linux-android`? 32-bit x86 android? is that really a thing?

hebasto commented:
@icota 
> I'm surprised `i686-linux-android` ABI is still supported. I would love to drop it...

@laanwj 
> What is `i686-linux-android`? 32-bit x86 android? is that really a thing?

Please consider #23744.

Making this PR a draft for now.

hebasto commented:
Updated 15676734ab90c9abe3c2e801197ba8045265263d -> ef81c51f62e03cb53e65f1ced6120d87f1827c40 ([pr23675.02](https://github.com/hebasto/bitcoin/commits/pr23675.02) -> [pr23675.03](https://github.com/hebasto/bitcoin/commits/pr23675.03)).

Dropped commits related to Android ABI as I'm going to implement them in a different way after #23744.

So this PR is ready to review again :tiger2:

hebasto commented:
#### Guix builds:
```
$ find guix-build-$(git rev-parse --short=12 HEAD)/output/ -type f -print0 | env LC_ALL=C sort -z | xargs -r0 sha256sum
dbad08c0e1d261a2b2099fd834b092fa232ec7ac890a560a07bb74a33003ec63  guix-build-ef81c51f62e0/output/aarch64-linux-gnu/SHA256SUMS.part
483045481812f827190c22cdc3783d766b1bb6a29c7061b5e65652aabd79b7d3  guix-build-ef81c51f62e0/output/aarch64-linux-gnu/bitcoin-ef81c51f62e0-aarch64-linux-gnu-debug.tar.gz
1db91a99bbf215628debf99d982aea12d0c772b6add541aecb2e6ca72e0773e4  guix-build-ef81c51f62e0/output/aarch64-linux-gnu/bitcoin-ef81c51f62e0-aarch64-linux-gnu.tar.gz
125dddb91a7e73dc3e9b6125cf9230351569e636bc9322053a352968fc94d90c  guix-build-ef81c51f62e0/output/arm-linux-gnueabihf/SHA256SUMS.part
d99e413fbfd0b74181a756d26a7a0d895d5d63c8a57002c2718affec2a4fd5c5  guix-build-ef81c51f62e0/output/arm-linux-gnueabihf/bitcoin-ef81c51f62e0-arm-linux-gnueabihf-debug.tar.gz
e4fe687ba5246302592be46bb1d68bdca14a02aec5115cd6b2a6a843c788129c  guix-build-ef81c51f62e0/output/arm-linux-gnueabihf/bitcoin-ef81c51f62e0-arm-linux-gnueabihf.tar.gz
6f3a9ffacc1e5e6bf30bd72b05f16821e5be3d874acdc19d226faa6ebefddfd4  guix-build-ef81c51f62e0/output/dist-archive/bitcoin-ef81c51f62e0.tar.gz
f05fedef4db71ea2a5c0e49fef86b6d896e11f4069ce6b4d38567afaa57355a3  guix-build-ef81c51f62e0/output/powerpc64-linux-gnu/SHA256SUMS.part
d6dd4dc6d9bd214f94c391b850d759967fad7d81003582be2171a635e6c67949  guix-build-ef81c51f62e0/output/powerpc64-linux-gnu/bitcoin-ef81c51f62e0-powerpc64-linux-gnu-debug.tar.gz
c867093fb3fe0d6cec097208d11e03f1606fe1a30e91efe0f8f56c0fc3149d95  guix-build-ef81c51f62e0/output/powerpc64-linux-gnu/bitcoin-ef81c51f62e0-powerpc64-linux-gnu.tar.gz
fb26d2d0058792af6e1f14bfedf69ecf7b4b7ca3af595977c28def36115e26d7  guix-build-ef81c51f62e0/output/powerpc64le-linux-gnu/SHA256SUMS.part
8be743747f113be713843846f0cc0a64abf5614c7406814ca17413f68a4d2e96  guix-build-ef81c51f62e0/output/powerpc64le-linux-gnu/bitcoin-ef81c51f62e0-powerpc64le-linux-gnu-debug.tar.gz
abdce4623c21d1ad88df151b40681ec38d2db6ac184e762bece02d4f89685529  guix-build-ef81c51f62e0/output/powerpc64le-linux-gnu/bitcoin-ef81c51f62e0-powerpc64le-linux-gnu.tar.gz
d9016d870232fa40618ebad47279b8fb33891d31fe98449ae9c2e2db2d490683  guix-build-ef81c51f62e0/output/riscv64-linux-gnu/SHA256SUMS.part
b1796d0a26cf38401596c651db5d581594bf9438c6e4f818b4fcbbb1d12a4e51  guix-build-ef81c51f62e0/output/riscv64-linux-gnu/bitcoin-ef81c51f62e0-riscv64-linux-gnu-debug.tar.gz
3810c9014e629912f2ca07d7635520ee2c661c11ac1673c5cd20e005015fe971  guix-build-ef81c51f62e0/output/riscv64-linux-gnu/bitcoin-ef81c51f62e0-riscv64-linux-gnu.tar.gz
176ede6148e4f0005cadea1efb5b09c1b500199c3c31efb7b01d9a92e7df9471  guix-build-ef81c51f62e0/output/x86_64-apple-darwin/SHA256SUMS.part
c3b1bb678ce5be90799412938f309c61e641d06899c23a6bcc4374fb9f5723b6  guix-build-ef81c51f62e0/output/x86_64-apple-darwin/bitcoin-ef81c51f62e0-osx-unsigned.dmg
b1bfe6344b4b2f34a0ad4a2d780335dcffda648fb8a5f31b2c73c26e77681593  guix-build-ef81c51f62e0/output/x86_64-apple-darwin/bitcoin-ef81c51f62e0-osx-unsigned.tar.gz
aa8a0243151b95e8263eeb2f5c92ab10e0d5a76ee14d52197e47a7ad128bc57f  guix-build-ef81c51f62e0/output/x86_64-apple-darwin/bitcoin-ef81c51f62e0-osx64.tar.gz
b979377080b34f6854b651fc52dbfe0290199d922edff6078fdf953c6cd12e29  guix-build-ef81c51f62e0/output/x86_64-linux-gnu/SHA256SUMS.part
9e3b4c74e9e1ee27aeff1a02102fd19a7a758d500a89ce763a482a45ce74d6be  guix-build-ef81c51f62e0/output/x86_64-linux-gnu/bitcoin-ef81c51f62e0-x86_64-linux-gnu-debug.tar.gz
f5c9a030ac5514c44f445003b9756b6a98172656c68b493560b43a2f6f74f4b4  guix-build-ef81c51f62e0/output/x86_64-linux-gnu/bitcoin-ef81c51f62e0-x86_64-linux-gnu.tar.gz
5593bb4867a2102cff4a9463e3efe91f4ab2681522cb2a404a00f5d5f89c3d63  guix-build-ef81c51f62e0/output/x86_64-w64-mingw32/SHA256SUMS.part
86163d4ea0439ca7de8630e562de8108d1bf7e7d0d203ff116de2bcb9eace928  guix-build-ef81c51f62e0/output/x86_64-w64-mingw32/bitcoin-ef81c51f62e0-win-unsigned.tar.gz
d13df6a9775f1d445a8c9ef441e30b6828a15fe8d251b000f6a3b2f506d87555  guix-build-ef81c51f62e0/output/x86_64-w64-mingw32/bitcoin-ef81c51f62e0-win64-debug.zip
5f8c829ede9c1d802a1de0b58b8cd0b7e4cf911b288bb82c2ca613bca59afa57  guix-build-ef81c51f62e0/output/x86_64-w64-mingw32/bitcoin-ef81c51f62e0-win64-setup-unsigned.exe
d43443b51ccc15c9266d483da84aa9a5a36d75570690fce83643ec15e4d93a52  guix-build-ef81c51f62e0/output/x86_64-w64-mingw32/bitcoin-ef81c51f62e0-win64.zip
```

fanquake reviewed (approved):
ACK ef81c51f62e03cb53e65f1ced6120d87f1827c40

It would be good if going forward these changes could be consolidated a bit more. #23744 probably could have been done here (once we'd decided that direction), rather than a new PR, given this is still making Android related changes? Over the last week there's been lot of PRs doing Android related things:
* https://github.com/bitcoin/bitcoin/pull/23675
* https://github.com/bitcoin/bitcoin/pull/23677
* https://github.com/bitcoin/bitcoin/pull/23678
* https://github.com/bitcoin/bitcoin/pull/23744
* https://github.com/bitcoin-core/gui/pull/504

and by the looks there's at least one more to go:
> Dropped commits related to Android ABI as I'm going to implement them in a different way after #23744.

Every new build related PR means more Guix builds, and often rebuilds, as these kinds of changes have a tendency to conflict with each other. It can also just be hard to keep track of what's being changed and why, especially when PRs are opened, then changed to do something else, and then a different PR is opened doing something similar/related.

Guix Build:
```bash
dbad08c0e1d261a2b2099fd834b092fa232ec7ac890a560a07bb74a33003ec63  guix-build-ef81c51f62e0/output/aarch64-linux-gnu/SHA256SUMS.part
483045481812f827190c22cdc3783d766b1bb6a29c7061b5e65652aabd79b7d3  guix-build-ef81c51f62e0/output/aarch64-linux-gnu/bitcoin-ef81c51f62e0-aarch64-linux-gnu-debug.tar.gz
1db91a99bbf215628debf99d982aea12d0c772b6add541aecb2e6ca72e0773e4  guix-build-ef81c51f62e0/output/aarch64-linux-gnu/bitcoin-ef81c51f62e0-aarch64-linux-gnu.tar.gz
125dddb91a7e73dc3e9b6125cf9230351569e636bc9322053a352968fc94d90c  guix-build-ef81c51f62e0/output/arm-linux-gnueabihf/SHA256SUMS.part
d99e413fbfd0b74181a756d26a7a0d895d5d63c8a57002c2718affec2a4fd5c5  guix-build-ef81c51f62e0/output/arm-linux-gnueabihf/bitcoin-ef81c51f62e0-arm-linux-gnueabihf-debug.tar.gz
e4fe687ba5246302592be46bb1d68bdca14a02aec5115cd6b2a6a843c788129c  guix-build-ef81c51f62e0/output/arm-linux-gnueabihf/bitcoin-ef81c51f62e0-arm-linux-gnueabihf.tar.gz
6f3a9ffacc1e5e6bf30bd72b05f16821e5be3d874acdc19d226faa6ebefddfd4  guix-build-ef81c51f62e0/output/dist-archive/bitcoin-ef81c51f62e0.tar.gz
f05fedef4db71ea2a5c0e49fef86b6d896e11f4069ce6b4d38567afaa57355a3  guix-build-ef81c51f62e0/output/powerpc64-linux-gnu/SHA256SUMS.part
d6dd4dc6d9bd214f94c391b850d759967fad7d81003582be2171a635e6c67949  guix-build-ef81c51f62e0/output/powerpc64-linux-gnu/bitcoin-ef81c51f62e0-powerpc64-linux-gnu-debug.tar.gz
c867093fb3fe0d6cec097208d11e03f1606fe1a30e91efe0f8f56c0fc3149d95  guix-build-ef81c51f62e0/output/powerpc64-linux-gnu/bitcoin-ef81c51f62e0-powerpc64-linux-gnu.tar.gz
fb26d2d0058792af6e1f14bfedf69ecf7b4b7ca3af595977c28def36115e26d7  guix-build-ef81c51f62e0/output/powerpc64le-linux-gnu/SHA256SUMS.part
8be743747f113be713843846f0cc0a64abf5614c7406814ca17413f68a4d2e96  guix-build-ef81c51f62e0/output/powerpc64le-linux-gnu/bitcoin-ef81c51f62e0-powerpc64le-linux-gnu-debug.tar.gz
abdce4623c21d1ad88df151b40681ec38d2db6ac184e762bece02d4f89685529  guix-build-ef81c51f62e0/output/powerpc64le-linux-gnu/bitcoin-ef81c51f62e0-powerpc64le-linux-gnu.tar.gz
d9016d870232fa40618ebad47279b8fb33891d31fe98449ae9c2e2db2d490683  guix-build-ef81c51f62e0/output/riscv64-linux-gnu/SHA256SUMS.part
b1796d0a26cf38401596c651db5d581594bf9438c6e4f818b4fcbbb1d12a4e51  guix-build-ef81c51f62e0/output/riscv64-linux-gnu/bitcoin-ef81c51f62e0-riscv64-linux-gnu-debug.tar.gz
3810c9014e629912f2ca07d7635520ee2c661c11ac1673c5cd20e005015fe971  guix-build-ef81c51f62e0/output/riscv64-linux-gnu/bitcoin-ef81c51f62e0-riscv64-linux-gnu.tar.gz
176ede6148e4f0005cadea1efb5b09c1b500199c3c31efb7b01d9a92e7df9471  guix-build-ef81c51f62e0/output/x86_64-apple-darwin/SHA256SUMS.part
c3b1bb678ce5be90799412938f309c61e641d06899c23a6bcc4374fb9f5723b6  guix-build-ef81c51f62e0/output/x86_64-apple-darwin/bitcoin-ef81c51f62e0-osx-unsigned.dmg
b1bfe6344b4b2f34a0ad4a2d780335dcffda648fb8a5f31b2c73c26e77681593  guix-build-ef81c51f62e0/output/x86_64-apple-darwin/bitcoin-ef81c51f62e0-osx-unsigned.tar.gz
aa8a0243151b95e8263eeb2f5c92ab10e0d5a76ee14d52197e47a7ad128bc57f  guix-build-ef81c51f62e0/output/x86_64-apple-darwin/bitcoin-ef81c51f62e0-osx64.tar.gz
b979377080b34f6854b651fc52dbfe0290199d922edff6078fdf953c6cd12e29  guix-build-ef81c51f62e0/output/x86_64-linux-gnu/SHA256SUMS.part
9e3b4c74e9e1ee27aeff1a02102fd19a7a758d500a89ce763a482a45ce74d6be  guix-build-ef81c51f62e0/output/x86_64-linux-gnu/bitcoin-ef81c51f62e0-x86_64-linux-gnu-debug.tar.gz
f5c9a030ac5514c44f445003b9756b6a98172656c68b493560b43a2f6f74f4b4  guix-build-ef81c51f62e0/output/x86_64-linux-gnu/bitcoin-ef81c51f62e0-x86_64-linux-gnu.tar.gz
5593bb4867a2102cff4a9463e3efe91f4ab2681522cb2a404a00f5d5f89c3d63  guix-build-ef81c51f62e0/output/x86_64-w64-mingw32/SHA256SUMS.part
86163d4ea0439ca7de8630e562de8108d1bf7e7d0d203ff116de2bcb9eace928  guix-build-ef81c51f62e0/output/x86_64-w64-mingw32/bitcoin-ef81c51f62e0-win-unsigned.tar.gz
d13df6a9775f1d445a8c9ef441e30b6828a15fe8d251b000f6a3b2f506d87555  guix-build-ef81c51f62e0/output/x86_64-w64-mingw32/bitcoin-ef81c51f62e0-win64-debug.zip
5f8c829ede9c1d802a1de0b58b8cd0b7e4cf911b288bb82c2ca613bca59afa57  guix-build-ef81c51f62e0/output/x86_64-w64-mingw32/bitcoin-ef81c51f62e0-win64-setup-unsigned.exe
d43443b51ccc15c9266d483da84aa9a5a36d75570690fce83643ec15e4d93a52  guix-build-ef81c51f62e0/output/x86_64-w64-mingw32/bitcoin-ef81c51f62e0-win64.zip
```
